### PR TITLE
Fork testing

### PR DESCRIPTION
**Pull request by Sidu28**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -30,8 +30,10 @@ jobs:
         run: forge test -vvv
         env:
           RPC_MAINNET: ${{ secrets.RPC_MAINNET }}
+          CHAIN_ID: ${{ secrets.CHAIN_ID }}
```
Comment (ChaoticWalrus) [inline .github/workflows/test.yml:33]: seems odd we're making the chain ID a secret?
I feel perhaps OK doing this for now but would like to see us get a better solution for this...

File: README.md [hunk 2]
```diff
@@ -46,6 +46,15 @@ The main command to run tests is:
 
 `forge test -vv`
 
+### Run Tests on a Fork
+Environment config is contained in config.yml.  Before running the following commands, [install yq](https://mikefarah.gitbook.io/yq/v/v3.x/).  Then set up the environment with this script:
+
+`source source-env.sh [CHAIN]`
```
Comment (ChaoticWalrus) [inline README.md:52]: so [CHAIN] can just be 'local' or 'goerli' right now?

File: src/test/EigenLayerDeployer.t.sol [hunk 3]
```diff
@@ -81,17 +81,39 @@ contract EigenLayerDeployer is Operators {
...
+    address beaconChainOracleAddress;
+    address emptyContractAddress;
+    address teamMultisig;
+    address communityMultisig;
+
+    uint256 goerliFork;
+
+
+
+
```
Comment (ChaoticWalrus) [inline src/test/EigenLayerDeployer.t.sol:111]: what is this variable for?

File: src/test/EigenLayerDeployer.t.sol [hunk 4]
```diff
@@ -119,7 +147,81 @@ contract EigenLayerDeployer is Operators {
...
+        ethPOSDeposit = new ETHPOSDepositMock();
+        pod = new EigenPod(ethPOSDeposit, delayedWithdrawalRouter, eigenPodManager, REQUIRED_BALANCE_WEI);
+
+        eigenPodBeacon = new UpgradeableBeacon(address(pod));
+
+        emit log("hehjr");
+
+
+
```
Comment (ChaoticWalrus) [inline src/test/EigenLayerDeployer.t.sol:179]: clean-up

File: src/test/EigenLayerDeployer.t.sol [hunk 5]
```diff
@@ -119,7 +147,81 @@ contract EigenLayerDeployer is Operators {
...
+        eigenStrat = StrategyBase(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(baseStrategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, eigenToken, eigenLayerPauserReg)
+                )
+            )
+        );
+        emit log_named_uint("strategy manager byes", address(strategyManager).code.length);
```
Comment (ChaoticWalrus) [inline src/test/EigenLayerDeployer.t.sol:217]: clean-up

File: .github/workflows/test.yml (hunk 1 above)
Comment (Sidu28) [reply .github/workflows/test.yml:33]: I think its ok for now since we don't want to run fork tests in CI.

File: README.md (hunk 2 above)
Comment (Sidu28) [reply README.md:52]: yup!

Comment (ChaoticWalrus) [reply README.md:52]: OK, probably should just say this in the README then

File: src/test/EigenLayerDeployer.t.sol (hunk 3 above)
Comment (Sidu28) [reply src/test/EigenLayerDeployer.t.sol]: deleted

File: src/test/unit/StrategyManagerUnit.t.sol [hunk 6]
```diff
@@ -527,13 +527,16 @@ contract StrategyManagerUnitTests is Test {
...
+        uint256 expiry = block.timestamp;
         bytes memory signature;
 
         {
             bytes32 structHash = keccak256(abi.encode(strategyManager.DEPOSIT_TYPEHASH(), strategy, token, amount, nonceBefore, expiry));
             bytes32 digestHash = keccak256(abi.encodePacked("\x19\x01", strategyManager.DOMAIN_SEPARATOR(), structHash));
 
+            emit log_named_bytes32("structHash", structHash);
+            emit log_named_bytes32("digestHash", digestHash);
+            emit log_named_bytes32("DOMAIN_SEPARATOR", strategyManager.DOMAIN_SEPARATOR());
```
[the remark this answers is not in the document]
Comment (Sidu28) [reply src/test/unit/StrategyManagerUnit.t.sol]: removed

File: README.md (hunk 2 above)
Comment (Sidu28) [reply README.md:52]: Added

**Review by ChaoticWalrus — CHANGES_REQUESTED** [review]
Couple minor nits which I've noted.
Overall I'm not a huge fan of this making an additional setup step necessary for running tests, but seems easy enough -- we'll see if we can work out a more ideal solution (like if Foundry adds nice support for custom flags) in the future.

**Comment by ChaoticWalrus** [comment]
@Sidu28 is there an easy way to improve what happens when you give an invalid chain name to the `source-env.sh` script? seems to kind of just hang rn.

**Comment by ChaoticWalrus** [comment]
For context: tons of tests are breaking when doing fork-testing on Goerli rn. This PR is minimal to get us to being able to run the tests at all; further changes will be necessary for tuning the tests to run appropriately against forked state.

**Comment by ChaoticWalrus** [comment]
I'm also seeing a few odd Foundry errors, which have been reported by @Sidu28 as well. The tests all appear to be running, so this is probably(?) OK. Screenshot attached:
screenshot

**Pull request merged**